curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/644b736c83c5250181633899.md

```diff
@@ -0,0 +1,234 @@
+---
+id: 644b736c83c5250181633899
+title: Step 65
+challengeType: 0
+dashedName: step-65
+---
+
+# --description--
+
+The `hasOwnProperty()` method is used to check if a specific property exists in an object. It returns `true` if the object has the property, otherwise `false`. For example, consider an object `person` with a property `name`:
+
+```js
+const person = {
+  name: 'John'
+};
+
+console.log(person.hasOwnProperty('name')); // Output: true
+console.log(person.hasOwnProperty('age')); // Output: false
+```
+
+You will create a `if` statement to check if the argument named `id` is available to the `categorieList` object. To do this task, you will need to use the `hasOwnProperty()` method.
+
+# --hints--
+
+You should use the `hasOwnProperty()` method to check if the `id` argument is available to the `categorieList` object.
+
+```js
+assert(code.match(/if\s*\(\s*categorieList\s*\.hasOwnProperty\s*\(\s*id\s*\)\s*\)/));
```

Comment:
```suggestion
assert(code.match(/if\s*\(\s*categoryList\s*\.hasOwnProperty\s*\(\s*(?:'id'|id)\s*\)\s*\)/));
```

Modified the regex to accept both arguments: `id` and `'id'`.

Reply:
properties are not strings in javascript objects